modules/xfmg/oas/sharedlib/xyna-openapi/generator-utils/src/main/java/com/gip/xyna/openapi/PrimitiveListTypeValidator.java

```diff
@@ -47,6 +71,8 @@ public void addValues(List<T> value) {
           newValidator.setValue(val);
           validators.add(newValidator);
         }
+      } else if (value == null) {
+          noList = true;
```

Comment:
It is a bit confusing, that addValues can lead to noList, because there can be values already there.
We should change the spelling of the method to setValue here and in XynaCodegenProperty and in the method body something like:
```
if (values==null) {
  validators = null;
} else {
  validators = new ArrayList<>();
  if (dummy != null) {
         for (T val : value) {
          V newValidator = dummy.get();
          newValidator.setName(getName());
          newValidator.setValue(val);
          validators.add(newValidator);
        }
}
```

Reply:
Code has been polished and your recommendations have been inserted. The helper variable noList has been removed in favor of introducing a method _getValidatorsNonNull_ to avoid an NPE in the generated code snippet.